File: 804. Unique Morse Code Words/804.go
package leetcode

func uniqueMorseRepresentations(words []string) int {
	morse := []string{".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."}
	res := make(map[string]bool, 0)

	for _,word := range words{
		tmp := ""
		for _, v := range word{
			tmp += morse[v-'a']
		}
		res[tmp] = true
	}
	return len(res)
}

File: 804. Unique Morse Code Words/_804.py
class Solution:
    def uniqueMorseRepresentations(self, words) -> int:
        morse = [".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."]
        res = set()
        for i in words:
            tmp = ""
            for j in i:
                tmp += morse[ord(j)-97]
            res.add(tmp)

        return len(res)

    def uniqueMorseRepresentations1(self, words):
        d = [".-", "-...", "-.-.", "-..", ".", "..-.", "--.", "....", "..", ".---", "-.-", ".-..", "--",
            "-.", "---", ".--.", "--.-", ".-.", "...", "-", "..-", "...-", ".--", "-..-", "-.--", "--.."]
        return len({''.join(d[ord(i) - ord('a')] for i in w) for w in words})